op-supervisor/supervisor/backend/backend.go

```diff
@@ -423,94 +423,94 @@ func (su *SupervisorBackend) DependencySet() depset.DependencySet {
-		safety, err := su.CheckMessage(ctx, msg.Identifier, msg.PayloadHash, executingDescriptor)
-		if err != nil {
-			su.logger.Error("Check message failed", "err", err,
-				"identifier", msg.Identifier, "payloadHash", msg.PayloadHash.String(), "executingTimestamp", executingDescriptor.Timestamp)
-			return fmt.Errorf("failed to check message: %w", err)
+	entries := inboxEntries
+	for {
+		if len(entries) == 0 {
+			break
 		}
-		if !safety.AtLeastAsSafe(minSafety) {
-			su.logger.Error("Message is not sufficiently safe",
-				"safety", safety, "minSafety", minSafety,
-				"identifier", msg.Identifier, "payloadHash", msg.PayloadHash.String(), "executingTimestamp", executingDescriptor.Timestamp)
-			return fmt.Errorf("message %v (safety level: %v) does not meet the minimum safety %v",
-				msg.Identifier,
-				safety,
-				minSafety)
+		remaining, acc, err := types.ParseAccess(entries)
+		if err != nil {
+			return fmt.Errorf("failed to read data: %w", err)
 		}
-	}
-	return nil
-}
-
-func (su *SupervisorBackend) CheckMessages(
-	ctx context.Context,
-	messages []types.Message,
-	minSafety types.SafetyLevel) error {
-	su.logger.Debug("Checking messages", "count", len(messages), "minSafety", minSafety)
+		entries = remaining
 
-	for _, msg := range messages {
-		su.logger.Debug("Checking message",
-			"identifier", msg.Identifier, "payloadHash", msg.PayloadHash.String())
-		// Guarantee message expiry checks do not fail by setting the executing timestamp to the message timestamp
-		// This is intentionally done to avoid breaking checkMessagesV1 which doesn't handle message expiry checks
-		safety, err := su.CheckMessage(ctx, msg.Identifier, msg.PayloadHash, types.ExecutingDescriptor{Timestamp: msg.Identifier.Timestamp})
+		msgBlock, err := su.checkAccess(ctx, acc, executingDescriptor)
 		if err != nil {
-			su.logger.Error("Check message failed", "err", err,
-				"identifier", msg.Identifier, "payloadHash", msg.PayloadHash.String())
-			return fmt.Errorf("failed to check message: %w", err)
+
 		}
-		if !safety.AtLeastAsSafe(minSafety) {
-			su.logger.Error("Message is not sufficiently safe",
-				"safety", safety, "minSafety", minSafety,
-				"identifier", msg.Identifier, "payloadHash", msg.PayloadHash.String())
-			return fmt.Errorf("message %v (safety level: %v) does not meet the minimum safety %v",
-				msg.Identifier,
-				safety,
-				minSafety)
+		// TODO add msgBlock to rewind lock
+
+		// TODO: this can be deferred to only check the latest block of all access entries
+		switch minSafety {
+		case types.LocalUnsafe:
+			err = nil // msg exists, nothing more to check
+		case types.CrossUnsafe:
+			err = su.chainDBs.IsCrossUnsafe(acc.ChainID, msgBlock)
+		case types.LocalSafe:
+			err = su.chainDBs.IsLocalSafe(acc.ChainID, msgBlock)
+		case types.CrossSafe:
+			err = su.chainDBs.IsCrossSafe(acc.ChainID, msgBlock)
+		case types.Finalized:
+			err = su.chainDBs.IsFinalized(acc.ChainID, msgBlock)
+		default:
+			err = types.ErrConflict
+		}
```

Comment:
Although we just removed `Safest`, could we move this to a ChainsDB function that accepts the type and does this case-switch internally? 

```
err = su.chainDBs.IsMinSafe(acc.ChainID, minSafety, msgBlock)
```

I like the way this API is directly checking if the minimum safety is supported, over how we used to grab the "safest" and compare, but this behavior could be hidden nicely.

Reply:
refactored it into a nice helper function